### ISSUE TITLE
clone_add_a_column has a bad argument check

### ISSUE THREAD

**Issue by pjcpjc** [ticdat/ticdat#184]
Look for the following code snippet.

```
current_fields = getattr(tdf_pdf, {"primary key": "primary_key_fields", "data": "data_fields"}[field_type])
verify(field not in current_fields, f"{field} already present in {current_fields}")
```

The `current_fields` will just be a dictionary mapping table name to fields, and not a list of the fields that the `field` argument shouldn't collide with. The fix is to instead have

```
current_fields = getattr(tdf_pdf, {"primary key": "primary_key_fields", "data": "data_fields"}[field_type])[table]
verify(field not in current_fields, f"{field} already present in {current_fields}")
```

This is unlikely to be a blocker for anyone, since anyone following `ticdat` naming conventions won't have the `field` argument collide with the buggy `current_fields`. Moreover, if `field` does collide with the correct implementation of `current_fields`, then that client programmer mistake will just result in a slightly more cryptic exception being thrown further down the line. 

In short, low priority bug, but worth fixing.

**Comment by pjcpjc** [comment]
Rats, as part of this bug  `verify(0 <= field_position <= len(current_fields)` line can block someone because `current_fields` might be too small. So be careful, because people might be using their own local version of `clone_add_a_column` and thus referring to `clone_a_anchillary_info_schema` directly. So don't change `clone_a_anchillary_info_schema` in the next release. Really, don't change that subroutine until 2025 sometime - just check its staying the same and then make another issue to check it again in 2025.
